pgscatalog.match/src/pgscatalog/match/lib/_arrow.py

```diff
@@ -45,8 +47,14 @@ def _(path: NormalisedScoringFile, tmpdir=None):  # type: ignore
         "row_nr": pl.UInt64,
     }
 
-    reader = pl.read_csv_batched(path.path, dtypes=dtypes, separator="\t")
-    return batch_read(reader, tmpdir=tmpdir, cols_keep=pl.all())
+    arrowpaths = []
+    for path in paths:
+        reader = pl.read_csv_batched(path.path, dtypes=dtypes, separator="\t")
+        arrowpaths.extend(
+            batch_read(reader, tmpdir=tmpdir, cols_keep=list(dtypes.keys()))
+        )
```

Comment:
This will still work with 1 scoring file that has multiple PGS (IDs) in it? Or multiple scoring files with multiple IDs? It seems like we might want to add a warning/error if the same PGS ID is found in multiple scoring files, no?

Reply:
With the new pydantic models we stopped supporting one scoring file with multiple PGS IDs (i.e. wide columns)

Adding a warning is a good idea! https://github.com/PGScatalog/pygscatalog/pull/62/commits/b5d18c2a52590985ab533cee45a97179bb54f306